yb-voyager/src/query/queryissue/parser_issue_detector_test.go

```diff
@@ -210,6 +219,11 @@ func TestAllIssues(t *testing.T) {
 		},
 	}
 
+	//Should modify it in similar way we do it actual code as the particular DDL issue in plpgsql can have different Details map on the basis of objectType
```

Comment:
I'm confused. What was wrong with the previous approach where you define issues directly with "FUNCTION", "list_high_earners" ?

Reply:
example for this cluster on -
```
func NewClusterONIssue(objectType string, objectName string, sqlStatement string) QueryIssue {
	details := map[string]interface{}{}
	//for ALTER AND INDEX both  same struct now how to differentiate which one to not
	if objectType == "TABLE" {
		details["INCREASE_INVALID_COUNT"] = false
	}
	return newQueryIssue(clusterOnIssue, objectType, objectName, sqlStatement, details)
}
``` 
earlier I was using the New function for getting queryissue - 
```
NewClusterONIssue("FUNCTION", "list_high_earners", "ALTER TABLE employees CLUSTER ON idx;")
```
giving this QueryIssue, with details as empty as - 
```
{{CLUSTER_ON ALTER TABLE CLUSTER not supported yet.  Remove it from the exported schema. https://github.com/YugaByte/yugabyte-db/issues/1124 https://docs.yugabyte.com/preview/yugabyte-voyager/known-issues/postgresql/#unsupported-alter-table-ddl-variants-in-source-schema map[]} FUNCTION list_high_earners ALTER TABLE employees CLUSTER ON idx; map[]}
```
But the actual query issue will look like this where details will be populated based no TABLE object type- 
```
{{CLUSTER_ON ALTER TABLE CLUSTER not supported yet.  Remove it from the exported schema. https://github.com/YugaByte/yugabyte-db/issues/1124 https://docs.yugabyte.com/preview/yugabyte-voyager/known-issues/postgresql/#unsupported-alter-table-ddl-variants-in-source-schema map[]} FUNCTION list_high_earners ALTER TABLE employees CLUSTER ON idx; map[INCREASE_INVALID_COUNT:false]}
```

So, to detect these issues in PLPGSQL, we modify the issues later (in getAllPLPGSQLIssues()) to change the objectType and objectName to the actual object name and type of the PLPGSQL object.
So now with cmp.Equal, it also compares `details` map I had to change the way I generate the expected issues to be able to check properly.

Reply:
Orthogonal point: 
`INCREASE_INVALID_COUNT` should not be in queryissue layer. 
It is an analyze-schema detail, we should have some logic of figuring it out at that layer.

Reply:
Yeah, I agree. I removed it in this PR with handling all cases for invalid count #2073